registration/tests/test_views.py

```diff
@@ -100,9 +100,26 @@ def _assert_registration_succeeds(self, form_data):
         with patch('registration.provision._provision_instance', autospec=True) as mock_handler:
             expected_call_count = 0
             for verification_email in mail.outbox:
-                verify_url = re.search(r'https?://[^\s]+',
-                                       verification_email.body).group(0)
-                self.client.get(verify_url)
+                # Extract validation code from email
+                verify_code = re.search(
```

Comment:
@giovannicimolin, this is ugly. [Obligatory reference](http://regex.info/blog/2006-09-15/247).
> Some people, when confronted with a problem, think “I know, I'll use regular expressions.”  Now they have two problems.

:stuck_out_tongue: 

But I don't think we can do anything else that is better here :man_shrugging: 
.

Reply:
I don't like this approach too (seems hacky to fetch the link from the email and run it through the browser).
Hopefully we get rid of this when doing e2e test of the new MFE.

Reply:
Do you have any suggestions on how to improve this?

Reply:
You can get the confirmation key from the `EmailAddress` model, generate the URL using the key and assert that the email contains the URL and also verify that the verification works using that URL. What do you say?

Reply:
Yup, definitively a better approach!